### PR TITLE
[NPU] Support minicpm-v with python cpp backend

### PR DESCRIPTION
**Pull request by plusbang**
## Description

Work with https://github.com/intel-analytics/llm.cpp/pull/756 to support minicpm-v models with python cpp backend.

### 2. User API changes

N/A

### 3. Summary of the change 
- add necessary padding operations for lm_head of minicpm_v_2_6 to avoid compile error.
- update hf_generate related logic to be compatible with `inputs_embeds` as input
- update cpp functions

### 4. How to test?
- [x] Application test
Details: https://github.com/analytics-zoo/nano/issues/1800#issuecomment-2566185398

### CODE REVIEW

File: python/llm/example/NPU/HF-Transformers-AutoModels/LLM/CPP_Examples/llm-npu-cli.cpp [hunk 1]
```diff
@@ -102,7 +102,7 @@ std::string run_generate(void* void_model, int32_t* embd_inp_ptr, int32_t embd_i
                          npu_model_params model_params, tokenizer_params tok_params, npu_generation_params generation_params, bool do_print){
     auto start = std::chrono::high_resolution_clock::now();
     float* logits = run_prefill(void_model, embd_inp_ptr, embd_inp_size,
-                                generation_params.repetition_penalty);
+                                generation_params.repetition_penalty, false);
```
Comment (rnwang04) [inline python/llm/example/NPU/HF-Transformers-AutoModels/LLM/CPP_Examples/llm-npu-cli.cpp:105]: Can we hide this false if it has a default value ?

**Review by rnwang04 — APPROVED** [review]
others LGTM.

File: python/llm/example/NPU/HF-Transformers-AutoModels/LLM/CPP_Examples/llm-npu-cli.cpp (hunk 1 above)
Comment (plusbang) [reply python/llm/example/NPU/HF-Transformers-AutoModels/LLM/CPP_Examples/llm-npu-cli.cpp]: > Can we hide this false if it has a default value ?

yes, have updated : )
